pkg/ingester/client/timeseries.go

```diff
@@ -41,3 +47,63 @@ func (p *PreallocTimeseries) Unmarshal(dAtA []byte) error {
 	p.Samples = make([]Sample, 0, expectedSamplesPerSeries)
 	return p.TimeSeries.Unmarshal(dAtA)
 }
+
+type LabelAdapter labels.Label
+
+func (bs *LabelAdapter) Marshal() ([]byte, error) {
+	buf := make([]byte, bs.Size())
+	_, err := bs.MarshalTo(buf)
+	return buf, err
+}
+
+func (bs *LabelAdapter) MarshalTo(buf []byte) (n int, err error) {
```

Comment:
Where is this function being used? Is this used by grpc/proto or somewhere else? Also needs comments.

Reply:
Will comment, for sure.  This is the implementation of the Protobuf encoding.